python/test/unit/nls/test_newton.py

```diff
@@ -100,23 +94,26 @@ def boundary(x):
         return np.logical_or(x[0] < 1.0e-8, x[0] > 1.0 - 1.0e-8)
 
     u_bc = fem.Function(V)
-    u_bc.vector.set(1.0)
-    u_bc.vector.ghostUpdate(addv=PETSc.InsertMode.INSERT, mode=PETSc.ScatterMode.FORWARD)
+    with u_bc.vector.localForm() as u_local:
```

Comment:
I would suggest that we set the boundary condition in the following way:
```
boundary_dofs = fem.locate_dofs_geometrical(V, boundary)
with u_bc.vector.localForm() as u_local:
    u_local.setValues(boundary_dofs, np.full(len(boundary_dofs), 1))
```
This limits the number of values we have to set in each boundary condition (and the function can be used for other constant DirichletBCs).
```

Reply:
Ignore this, as it will not be a problem when we introduce `dolfinx.Constant` as an input argument to `DirichletBC`